### PR TITLE
Make package installable via `pip`, support gradient accumulation, and max norm-clip correctly for FSDP

### PR DESCRIPTION
**Pull request by amorehead**
* Makes the package installable via `pip`
* Ensures `ZClipLightningCallback` steps right at `on_before_optimizer_step()` (and therefore after all calls to `on_after_backward()`) to ensure gradients are (optionally) accumulated correctly
* Globally clips gradient norms by a specified maximum norm value correctly using rank syncing (see https://docs.pytorch.org/docs/stable/fsdp.html#torch.distributed.fsdp.FullyShardedDataParallel.clip_grad_norm_)

### CODE REVIEW

**Comment by amorehead** [comment]
Also, on a different note, there's an MIT license statement at the bottom of the `README.md`, but the actual license attached to this repository is an Apache-2 license?

**Comment by amorehead** [comment]
Lastly, does [this line](https://github.com/amorehead/ZClip/blob/bcdab02154e74e66779d75d3a74d75914e65aa03/zclip.py#L177) accidentally accumulate the total gradient norm for each (e.g., GPU) rank instead of once for all ranks? If the former is the case, this would lead to the warmup being performed incorrectly (i.e., being much shorter than intended).

**Comment by louisowen6** [comment]
> Also, on a different note, there's an MIT license statement at the bottom of the `README.md`, but the actual license attached to this repository is an Apache-2 license?

Hi @amorehead , thank for your PR, it's being reviewed by our team. 

The correct license is Apache-2 License, sorry for the confusion :)
